tidy3d/components/tcad/simulation/heat_charge.py

```diff
@@ -1778,3 +1786,129 @@ def _useHeatSourceFromConductionSim(self):
+                    # the following dictionary associates a structure with a modified medium
+                    struct_medium = {}
+                    for structure in self.structures:
+                        if isinstance(structure.medium.charge, SemiconductorMedium):
+                            sc_medium = copy.deepcopy(structure.medium.charge)
+
+                            dopants = {"N_a": sc_medium.N_a, "N_d": sc_medium.N_d}
+                            for key, dopant in dopants.items():
+                                if isinstance(dopant, SpatialDataArray):
+                                    new_dopant = (
+                                        dopant.rename({"z": "z_tmp"})
+                                        .rename({["x", "y"][zero_dims[0]]: "z"})
+                                        .rename({"z_tmp": ["x", "y"][zero_dims[0]]})
+                                    )
+                                    sc_medium = sc_medium.updated_copy(**{key: new_dopant})
+                                elif isinstance(dopant, tuple):
+                                    new_boxes = []
+                                    for doping_box in dopant:
+                                        if isinstance(doping_box, GaussianDoping):
+                                            new_center = list(doping_box.center)
+                                            new_center[zero_dims[0]] = doping_box.center[2]
+                                            new_center[2] = doping_box.center[zero_dims[0]]
+
+                                            new_size = list(doping_box.size)
+                                            new_size[zero_dims[0]] = doping_box.size[2]
+                                            new_size[2] = doping_box.size[zero_dims[0]]
+
+                                            source = doping_box.source
+                                            if "z" in source:
+                                                source = source.replace(
+                                                    "z", ["x", "y"][zero_dims[0]]
+                                                )
+
+                                            new_boxes.append(
+                                                doping_box.updated_copy(
+                                                    center=new_center, size=new_size, source=source
+                                                )
+                                            )
+                                        else:
+                                            new_boxes.append(doping_box)
+                                        sc_medium = sc_medium.updated_copy(**{key: new_boxes})
+                            struct_medium[structure] = structure.medium.updated_copy(
+                                charge=sc_medium
+                            )
+                        else:
+                            struct_medium[structure] = structure.medium
```

Comment:
let's also invert here:
```
if not isinstance(structure.medium.charge, SemiconductorMedium):
    struct_medium[structure] = structure.medium
else:
    ...
```
otherwise hard to remember what this `else` was for